test/integration/filters/stop_iteration_and_continue_filter.cc

```diff
@@ -64,6 +65,9 @@ class StopIterationAndContinueFilter : public Http::PassThroughFilter {
     if (end_stream) {
       setEndStreamAndDecodeTimer();
     }
+    if (stop_and_buffer_) {
+      return Http::FilterDataStatus::StopIterationAndBuffer;
```

Comment:
If this change is not added, [assertion](https://github.com/envoyproxy/envoy/blob/main/source/common/tcp_proxy/tcp_proxy.cc#L760) fails in tcp_proxy.cc 
```
 // The upstream should consume all of the data.
  // Before there is an upstream the connection should be readDisabled. If the upstream is
  // destroyed, there should be no further reads as well.
  ASSERT(0 == data.length());
```

Why we dont need this assertion in HCM case and it is there only in tcp_proxy?

Reply:
StopIterationNobuffer is generally called before we quit processing a request.  it results in discarding data that's in the filter.
in the tcp proxy case it means the filter manager doesn't consume the data to